overlord/snapstate/snapmgr.go

```diff
@@ -130,6 +131,9 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	st.Lock()
 	snapst.Candidate = &snap.SideInfo{}
+	if ss.Flags&int(snappy.DeveloperMode) != 0 {
+		snapst.DevMode = true
+	}
```

Comment:
this seems too early, should be done when the snap becomes active (when we set snapst.Active = true), with matching undo code